chainer/link.py

```diff
@@ -211,9 +272,27 @@ def add_persistent(self, name, value):
             raise AttributeError(
                 'cannot register a new persistent value %s: attribute exists'
                 % name)
-        self._persistent.append(name)
+        self._persistent.add(name)
+        self._params.discard(name)
         d[name] = value
 
+    def register_persistent(self, name):
+        """Registers an attribute of a given name as a persistent value.
+
+        This is a convenient method to register an existing attribute as a
+        persistent value.
+
+        Args:
+            name (str): Name of the attribute to be registered.
+
+        """
```

Comment:
I think it is more kind to write to the document that if `name` is already registered as a parameter (i.e. in `self._params`), this method  "hijacks" the corresponding parameter, that is, unregisters it from `_params` and reregisters it as a persistent value.